### PR TITLE
mimic node more closely

### PR DESCRIPTION
**Pull request by LinusU**
This is the fix first suggested by @crispy1989 in #46 but tweaked to work better and have correct tests.

> Node's default behavior is to only print an exception and exit if there is no other uncaughtException handler registered. This emulates that default behavior by bypassing the uncaughtException handler if other handlers are registered.

It also includes #76 so **pull that before** pulling this and this pull request should automatically be updated to only show the one relevant commit.

@evanw @julien-f If you want another maintainer I would be willing to help :)

### CODE REVIEW

**Comment by LinusU** [comment]
Test passes in browser! Good to merge :+1:

**Comment by julien-f** [comment]
Please rebase on master to avoid multiple *support charset...* commits.

**Comment by LinusU** [comment]
Done :)

**Comment by julien-f** [comment]
I find your implementation a bit complicated, why did you not simply change `handleUncaughtException()` to check for a single listener (itself)?

**Comment by LinusU** [comment]
You could read some of the reasoning from when I did a similar thing before here: segmentio/oh-crap#1

**tl;dr** If another module would to the same thing as us, it would break both.

**Comment by julien-f** [comment]
I am not a fan of redefining `process.emit()` but I see your point.

**Comment by julien-f** [comment]
The only thing that worries me is that we have no guarantee that Node will use `process.emit()` to trigger event, it may use an internal version or even the original definition of `emit()`.

**Comment by LinusU** [comment]
Hmm, yeah I see your point. Well, I've read the source code and I know that it works for now. I don't see it as such a big risk thought, if that happens we can always fix it then.

**Comment by julien-f** [comment]
Thanks.
